src/main/java/org/springframework/data/couchbase/core/ExecutableFindByIdOperationSupport.java

```diff
@@ -63,13 +62,13 @@ public T one(final String id) {
 		}
 
 		@Override
-		public TerminatingFindById<T> inCollection(final String collection) {
+		public ExecutableFindById<T> inCollection(final String collection) {
 			Assert.hasText(collection, "Collection must not be null nor empty.");
 			return new ExecutableFindByIdSupport<>(template, domainType, collection, fields);
 		}
 
 		@Override
-		public FindByIdWithCollection<T> project(String... fields) {
+		public ExecutableFindById<T> project(String... fields) {
 			Assert.notEmpty(fields, "Fields must not be null nor empty.");
```

Comment:
I find it strange that FindById does projection by project(fields) instead of as(returnType) like everything else.